### PR TITLE
doc: Update documentation for device JSON objects

### LINKED ISSUE #29: Update device JSON format documentation
#### Summary

https://thethingsstack.io/getting-started/migrating-from-networks/device-json/ contains misleading and/or plain wrong info.

#### What is missing? What do you want to see?

Improve misleading sections.

#### How do you propose to document this?

- Rename `Required Fields ...` to `Fields ...`, as optional fields are listed as well.
- `ids.application_id` field is wrong -- also add a note that it is not required and will be overwritten
- For LoRaWAN versions, regional parameters versions and Frequency Plans, link to the appropriate doc pages. The current link to the term glossary is not helpful.
- `root_keys.nwk_app.key` and `root_keys.nwk_key.key` should be `app_key` and `nwk_key`. Also add note that `nwk_key` is only needed for LoRaWAN 1.1+. Fix inconsistencies throughout the doc.
- Add examples for devices with/without active session, possibly for different versions as well. We could have downloadable example json files (like we do elsewhere with manifest files). 

#### Can you do this yourself and submit a Pull Request?

Assigning to me, will re-assign if needed.

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #29

#### Changes

- Add proper examples for OTAA devices and OTAA devices with existing sessions
- Improve field descriptions, add links for values of enum fields.
- Specify which fields are required and under which conditions.
- Add example value for each field

#### Notes for Reviewers

Most of the changes are based on customer feedback.

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Run Locally: Verified that the docs build using `make server`
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/getting-started/migrating-from-networks/device-json.md [hunk 1]
```diff
@@ -6,93 +6,147 @@ weight: 1
...
-    },
-    "..."
+    "last_f_cnt_up": 32,
+    "last_n_f_cnt_down": 10
   }
 }
 ```
 
-For more information on configuring MAC settings, see [Fine-tuning MAC Settings]({{< ref "getting-started/migrating-from-v2/configure-mac-settings" >}}).
+> *Note*: For more information on configuring MAC settings, see [Fine-tuning MAC Settings]({{< ref "getting-started/migrating-from-v2/configure-mac-settings" >}}).
```
Comment (benolayinka) [inline doc/content/getting-started/migrating-from-networks/device-json.md:152]:
```suggestion
> **Note**: For more information on configuring MAC settings, see [Fine-tuning MAC Settings]({{< ref "getting-started/migrating-from-v2/configure-mac-settings" >}}).
```

File: doc/content/getting-started/migrating-from-networks/device-json.md [hunk 2]
```diff
@@ -6,93 +6,147 @@ weight: 1
...
+]
+```
+
+## JSON End Device Format
+
+The full specification of the JSON format is defined in the API protos, see the [EndDevice]({{< ref "/reference/api/end_device#message:EndDevice" >}}) message definition for details.
+
+The linked specification is quite extensive, and contains a lot of fields that are not required, or are only set and used internally by the Network Server. Below, the required and most commonly used fields are discussed.
+
+| Field | Required | Type | Example | Description |
```
Comment (benolayinka) [inline doc/content/getting-started/migrating-from-networks/device-json.md:51]: This table overflows the content column and isn't scrollable :( Plz create an issue to fix it and I'll look at it when I have time.

localhost_1313_getting-started_migrating-from-networks_device-json_(Laptop with HiDPI screen)

**Review by benolayinka — APPROVED** [review]

File: doc/content/getting-started/migrating-from-networks/device-json.md [hunk 3]
```diff
@@ -6,93 +6,147 @@ weight: 1
...
+    "app_key": {
+      "key": "01020304050607080102030405060708"
+    }
+  }
+}
+```
+
+</summary>
+
+<summary><description>Example OTAA device with existing session:</description>
```
Comment (laurensslats) [inline doc/content/getting-started/migrating-from-networks/device-json.md:110]:
```suggestion
<summary><h4>Example OTAA device with existing session:</h4>
```

File: doc/content/getting-started/migrating-from-networks/device-json.md [hunk 4]
```diff
@@ -6,93 +6,147 @@ weight: 1
...
 >   }
-> }, 
+> }
 > ```
 
-## Example `devices.json`
+## Examples
 
-Below is an example `devices.json` file. The file may contain multiple devices, stored as different JSON objects.
+<summary><description>Example OTAA device:</description>
```
Comment (laurensslats) [inline doc/content/getting-started/migrating-from-networks/device-json.md:85]:
```suggestion
<summary><h4>Example OTAA device:</h4>
```

**Review by laurensslats — APPROVED** [review]
Please update the 2 comments I made to make a piece of text better readable. 

Rest looks good.

**Pull request merged**